### PR TITLE
Deleting received messages

### PR DESCRIPTION
**Pull request by aniansson**
## Changes proposed in this pull request:
Added tests for deleting a received message

## What I have learned working on this feature: 
How to write tests for having content in mailbox and how to delete them. And how to nest steps.

Ready for review!

### CODE REVIEW

File: features/deleting.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Feature: As a visitor,
+  I would like to delete emails,
+  that are receiven
+
+  Background:
+    Given there is users named:
```
Comment (AmberWilkie) [inline features/deleting.feature]: Isn't there already a step like `Given the following users exist`? Reuse that step instead of writing new ones.

**Comment by tochman** [comment]
The title: `Deleting feature`. Deleting what?

File: features/step_definitions/deleting_steps.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+Given(/^there is users named:$/) do |table|
+  table.hashes.each do |hash|
+    FactoryGirl.create(:user, hash)
+  end
+end
+
+Given(/^I send an email to "([^"]*)"$/) do |name|
```
Comment (AmberWilkie) [inline features/step_definitions/deleting_steps.rb]: This is not a blocker, but there are helper methods in the "tips and tricks" section that you could use instead of going through the site in your test. See if you can't refactor this to use the "I send a message" helper.

**Review by AmberWilkie — COMMENTED** [review]
Good work.

File: features/deleting.feature [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Feature: As a visitor,
+  I would like to delete emails,
+  that are receiven
```
Comment (tochman) [inline features/deleting.feature]: This user story doesn't feel very well defined.

File: features/deleting.feature (hunk 1 above)
Comment (tochman) [inline features/deleting.feature]: `there is users named:` Hmm... doesn't feel like a good step name

File: features/deleting.feature [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  Background:
+    Given there is users named:
+      | name      | email            | password |
+      | Test-User | email@random.com | password |
+      | User1     | user1@random.com | password |
+
+  Scenario: Deleting a message
+    Given "Test-User" is logged in
+    And I send an email to "User1"
+    And I should be on the "index" page
```
Comment (tochman) [inline features/deleting.feature]: `I should be` feels like an assertion. I'm getting a feeling that you are navigating to the `index` page, right? Please revisit this and rephrase.

File: features/deleting.feature [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    And I click "Logout"
+    Given "User1" is logged in
```
Comment (tochman) [inline features/deleting.feature:15]: Avoid using `Given` when an `And` should do. Also, I understand that you are reusing a step definition (perfectly fine) but `"User1" is logged in` doesn't feel right here. Can you rephrase?

File: features/step_definitions/deleting_steps.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+Given(/^there is users named:$/) do |table|
```
Comment (tochman) [inline features/step_definitions/deleting_steps.rb]: See comment above

**Review by tochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/basic_steps.rb [hunk 7]
```diff
@@ -7,7 +7,7 @@
   expect(page).to have_content content
 end
 
-Given(/^that I'm not logged in$/) do
+Given(/^I log out$/) do
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]:
```
Given(/^(?:that user logs out|I log out)$/) do
```

File: features/step_definitions/deleting_steps.rb [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  table.hashes.each do |hash|
+    FactoryGirl.create(:user, hash)
+  end
+end
+
+Given(/^"([^"]*)" sends a message to "([^"]*)"$/) do |sender, name|
+user = User.find_by(name: sender)
+receiver = User.find_by(name: name)
+user.send_message(receiver, "body", "content")
+end
```
Comment (tochman) [inline features/step_definitions/deleting_steps.rb:11]: Kod indrag

**Review by tochman — APPROVED** [review]
